libs/parallelism/algorithms/include/hpx/parallel/algorithms/for_each.hpp

```diff
@@ -376,8 +376,59 @@ namespace hpx { namespace parallel { inline namespace v1 {
             for_each_iteration& operator=(for_each_iteration&&) = default;
 
             template <typename Iter>
-            HPX_HOST_DEVICE HPX_FORCEINLINE void operator()(Iter part_begin,
-                std::size_t part_size, std::size_t /*part_index*/)
+            HPX_HOST_DEVICE HPX_FORCEINLINE void operator()(
+                Iter part_begin, std::size_t part_size,
+                std::size_t /*part_index*/)
+            {
+                hpx::util::annotate_function annotate(f_);
+                execute(part_begin, part_size);
+            }
+        };
+
+        template <typename ExPolicy, typename F>
+        struct for_each_iteration<ExPolicy, F, util::projection_identity>
+        {
+            using execution_policy_type = typename std::decay<ExPolicy>::type;
```

Comment:
Not necessary to change for this PR, but since we're requiring C++14 support I've started using `decay_t` (and other `_t`s where available).

Reply:
Yes, I think we should do that in a big sweep.